Cheat: Hold Drawbridge at Night

Dumbledork01 opened the pull request:
**Cheat**
* Hyrule Castletown drawbridge remains open throughout the night. I added this for convenience of young link players.

**Demonstration Videos**
Drawbridge Cheat: https://youtu.be/sHCkYhPQr6M

Dumbledork01 committed:
Added cheat to hold drawbridge

If the player selects the cheat to keep the drawbridge open, the drawbridge will not update when night comes.
Added cheat for Drawbridge
Added a cheat that will keep the drawbridge open at night.

aMannus commented:
I don't know if you're willing to look into it, but it might be nice to also have the bridge open from hyrule castle town's side when this enhancement is enabled.

Dumbledork01 commented:
@aMannus I was thinking about adding that last night when I made this PR, but I was too tired haha. I'll look into it because that would be very helpful!

Dumbledork01 committed:
Bridge Spawns as open

Dumbledork01 committed:
Leave castle town at night (young link)

Added logic to allow young link to leave. It sets the time to day while he is near the bridge, then sets it back to night when he exits either into the pot house, the market, or Hyrule field.

Dumbledork01 committed:
Fixed typo

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Fixed issues

Bridge now will properly lower during opening cut scene.
Since exiting the marketplace at night is only able to be done if it is rendered as daytime, that pre-rendered scene is the daytime variant. However, to accommodate this, I've disabled back ground music so it's less jarring of a transition.

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Merge branch 'HarbourMasters:develop-zhora' into Hold-Drawbridge-Cheat

Dumbledork01 committed:
Made leaving Nighttime castletown more seamless

briaguya0 commented:
similar functionality has now been implemented via https://github.com/HarbourMasters/Shipwright/pull/2128

i'm going to close this, but thank you so much for making the PR!
